### PR TITLE
Add advanced search options [SCI-1158]

### PR DESCRIPTION
**Pull request by okriuchykhin**
https://biosistemika.atlassian.net/browse/SCI-1158

### CODE REVIEW

File: app/assets/javascripts/search.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+$(document.body).ready(function() {
```
Comment (ZmagoD) [inline app/assets/javascripts/search.js]: Can you put this in an IIFE? 
(function() {
  'use strict';
  ....
  ....
})();

File: app/controllers/search_controller.rb [hunk 2]
```diff
@@ -41,47 +39,60 @@ def new
   private
 
   def load_vars
-    @search_query = params[:q] || ''
+    query = params[:q].strip || ''
```
Comment (ZmagoD) [inline app/controllers/search_controller.rb]: you can use 'fetch' to set the default value here if you want:

query = params.fetch(:q) { '' }

File: app/models/asset.rb [hunk 3]
```diff
@@ -152,11 +179,12 @@ def self.search(
...
     Asset
-      .joins("LEFT JOIN asset_text_data ON assets.id = asset_text_data.asset_id")
-      .select("assets.*")
-      .select("ts_headline(data, to_tsquery('" + s_query + "'),
-        'StartSel=<mark>, StopSel=</mark>') headline")
-      .where("assets.id IN (?)",  ids)
+      .joins('LEFT JOIN asset_text_data ON ' \
+             ' assets.id = asset_text_data.asset_id')
+      .select('assets.*')
+      .select("ts_headline(data, to_tsquery('" + s_query +
```
Comment (ZmagoD) [inline app/models/asset.rb]: Can you use ActiveRecord::Sanitization for s_query variable here or sanitize it before you do the trim/map/join stuff above?

File: app/assets/javascripts/search.js (hunk 1 above)
Comment (okriuchykhin) [inline app/assets/javascripts/search.js]: Done

File: app/models/asset.rb (hunk 3 above)
Comment (okriuchykhin) [inline app/models/asset.rb]: done

File: app/controllers/search_controller.rb (hunk 2 above)
Comment (okriuchykhin) [inline app/controllers/search_controller.rb]: done

**Comment by ZmagoD** [comment]
Good job! :rocket:

**Pull request merged**